### ISSUE TITLE
Timer in the app when entering carbs (Feature request)

### ISSUE THREAD

**Issue by darmardesign** [Artificial-Pancreas/iAPS#322]
Hi all,

I have been thinking recently. Can we get timer in bolus menu ? I need 20 minutes pre-bolus so i always need to open timer on my phone and enter it. Is there somehow possibility that we integrate it and make custom buttons for time (My case 15-20-35 min) so when we add bolus it activats?

Thank you

**Comment by Jon-b-m** [comment]
I guess.. But were exactly?

**Comment by darmardesign** [comment]
Thank you for reply.

In bolus menu, somewhere at the bottom? as we give bolus, it turns on or we turn it on.

**Comment by flyingpie101** [comment]
He is asking for a Notification/Alarm/Reminder to start to eat 

My daughter often does that as well.  Pre-bolus for a snack, then sets a Timer to go off in 15 mins to remind her to eat

**Comment by avouspierre** [comment]
OK ! The best is to use shortcuts to do this : 
![IMG_24FC365A8F37-1](https://github.com/Artificial-Pancreas/iAPS/assets/4339604/f2a2da5a-09b9-4e19-8443-29c3d530e034)

**Comment by flyingpie101** [comment]
But that doesnt give you a pre-bolus does it ?  I just see carbs being entered

**Comment by avouspierre** [comment]
You can replace with a temp basal (pre-bolus) function

**Comment by flyingpie101** [comment]
of course you can, however, this scenario doesnt work out.

in iAPS:
Pre-bolus, by entering carbs 
Get a calculated bolus
Bolus
Set a timer to eat 

I assume the shortcut cannot get a recommended bolus based on a carb entry ?

**Comment by avouspierre** [comment]
Yes but use the smb algo doing the job to inject the Bolus in the next minutes.. but if you use the recommended bolus not possible currently.
